docs/source/data_structures.md

```diff
@@ -0,0 +1,138 @@
+
+* **Select**
+  Filtering datasets by dimensions such as country, sector, or gas is essential. The `primap2` `.loc` function already supports this.
+
+* **Set**
+  Assigning values to datasets across dimensions is another critical task.
+
+* **Downscale**
+  The ability to downscale datasets is necessary.
+
+We aim to retain as much of the current codebase as possible, though some parts may require rewriting.
+
+#### Possible approaches
+
+* We considered several approaches and chose [xarray datatree](https://docs.xarray.dev/en/stable/generated/xarray.DataTree.html).
+* Excluded approaches:
+  * [**Sparse array on top of numpy**](https://sparse.pydata.org/en/stable/): Memory-efficient but slow, with limited function optimization for our needs. Merge operations do not scale well. Some experiments with sparse arrays [here](https://github.com/primap-community/primap2/pull/315/files).
+  * **SQL database**: Could serve as a backend for data storage but is too slow for calculations and not suitable for our workflows.
+  * **Pandas**: Useful for pre-processing, where we already use it, but rewriting everything in pandas would be a significant effort. Mathematical operations are slow, and datasets remain sparse after converting back to xarray. Basiv example of conversion in pandas [here](https://github.com/primap-community/primap2/pull/321/files).
+
+#### DataTree
+
+* Data trees are a hierarchical structure of datasets.
+* They offer a data structure for managing multiple datasets without the need to merge them (which is slow for sparse arrays).
+* DataTree replaces looping through datasets to perform operations by providing a hierarchical tree of datasets.
+* Regarding our sparsity issue: combining several datasets in one DataTree helps to some extent but does not eliminate the problem. For example, a unique category in dataset A would not create all NaNs for that dimension in dataset B but would still create many NaNs in dataset A.
+* It is relatively easy to convert multiple datasets into an xarray DataTree.
+* Watch this [video](https://www.youtube.com/watch?v=Iwcuy6Smbjs) for an introduction to datatree
+
+<table>
+  <tr>
+    <th>Pro</th>
+    <th>Con</th>
+  </tr>
+  <tr>
+    <td>
+      <ul>
+        <li>Supports conversion from/to NetCDF and Zarr.</li>
+        <li>Allows us to stay within the xarray ecosystem (as opposed to SQL or pandas).</li>
+        <li>All pre-processing steps can be covered in DataTree.</li>
+      </ul>
+    </td>
+    <td>
+      <ul>
+        <li>Limited examples and documentation available (as of April 2025).</li>
+        <li>Requires making our primap2 functions compatible with DataTree.</li>
+        <li>Processing DataTrees may involve looping through sub-datasets. Is it faster than merging once?</li>
+        <li>The hierarchical concept does not align well with our use case. All country datasets would be "siblings," but this is not necessarily a problem.</li>
+      </ul>
+    </td>
+  </tr>
+</table>
```

Comment:
```suggestion
| Pro | Con |
|---|---|
| Supports conversion from/to NetCDF and Zarr. | Limited examples and documentation available (as of April 2025). |
| Allows us to stay within the xarray ecosystem (as opposed to SQL or pandas). | Requires making our primap2 functions compatible with DataTree. |
| All pre-processing steps can be covered in DataTree.  | Processing DataTrees may involve looping through sub-datasets. Is it faster than merging once? |
| | The hierarchical concept does not align well with out use case. All country datasets would "siblings," but this is not necessarily a problem. |
```

It's not exactly the same result, but if possible avoid html and use markdown instead.